### ISSUE TITLE
Observing consistently lower performance scores when running Lighthouse programmatically than from Chrome DevTools

### ISSUE THREAD

**Issue by wolvriene** [GoogleChrome/lighthouse#10062]
Settings Used in both Programmatically  and DevTool audit are same and are listed below .
Chrome Version : 78.0.3904.108
Node.js version: 13.1.0
Device :Desktop
Audits: Performance
Throttling : No Throttling(Provided by environment)
Other details provided by report:
**User agent (host)**: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/78.0.3904.108 Safari/537.36
**User agent (network):** Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3694.0 Safari/537.36 Chrome-Lighthouse

Ran audit on Devtool and by Programmatically .Results are different .
**Programmatic Performance Score**: 22
First Contentful Paint
1.7 s
Speed Index
4.0 s
Time to Interactive
6.5 s
First Meaningful Paint
4.4 s
First CPU Idle
6.5 s
Max Potential First Input Delay
280 ms

**DevTools Performance Score:** 82
Metrics
First Contentful Paint
0.7 s
Speed Index
2.8 s
Time to Interactive
5.0 s
First Meaningful Paint
5.0 s
First CPU Idle
5.0 s
Max Potential First Input Delay
480 ms
 Please revert if any other additional information is required .

**Comment by patrickhulce** [comment]
How are you setting "no throttling" programmatically? Are you disabling the storage reset in DevTools?

**Comment by wolvriene** [comment]
Hi @patrickhulce 
i have set "no throttling " programatically as below
```
       disableNetworkThrottling: true,
       disableCpuThrottling: true,
       throttlingMethod: 'provided',

```
Also disabled the storage reset in DevTools as "Clear storage"
attached in screenshots.
Please let me know for any changes 
<img width="690" alt="Screenshot 2019-12-02 at 11 51 08 PM" src="https://user-images.githubusercontent.com/58415776/69984366-00570200-155f-11ea-84b0-20fce3c33339.png">

**Comment by patrickhulce** [comment]
Hm, that is how you set no throttling in programmatic so everything looks good there (just `throttlingMethod: 'provided'` should be all that's needed). The only other thing that seems to be different is headless vs. headful Chrome.

Is this URL public for us to reproduce the issue?

**Comment by wolvriene** [comment]
@patrickhulce 
I checked with both headless and headful Chrome .
There is no much difference, Observing consistently lower performance scores even in headful (Programmatically)

URLs are not public

**Comment by wolvriene** [comment]
Hi @patrickhulce 
Any update on this ?
Thanks

**Comment by patrickhulce** [comment]
Wait @wolvriene something is not right with the numbers you've reported. You shouldn't be getting anywhere near a 22 with [those metric values](https://paulirish.github.io/lh-scorecalc/).

![image](https://user-images.githubusercontent.com/2301202/70253891-b54d1280-1749-11ea-85e4-5bd17b82a20f.png)

In fact the difference in the metric values themselves is pretty well within the range of reasonable [variance](https://github.com/GoogleChrome/lighthouse/blob/master/docs/variability.md) for different Chrome variants

**Comment by wolvriene** [comment]
@patrickhulce :
Please find a different screenshot with similar kind of number in which i am getting score as 22%.
I tried with [lh-scorec-calc](https://paulirish.github.io/lh-scorecalc/) getting score 71%, is there any setting/configuration need to change ?


<img width="1185" alt="Screenshot 2019-12-09 at 3 47 16 PM" src="https://user-images.githubusercontent.com/58415776/70427882-86ad9f80-1a9b-11ea-8247-adce985ef7d2.png">

**Comment by patrickhulce** [comment]
Oooooh, is your version of programmatic later than the DevTools version of Lighthouse?

Newer Lighthouse will automatically adjust the scores to desktop curves while older stable DevTools does not yet do this. This would explain the score discrepancy.

**Comment by wolvriene** [comment]
@patrickhulce 
 i am using Programmatic lighthouse version: 5.6.0

How to check lighthouse DevTool version ? 
i am using Chrome Version : 78.0.3904.108

**Comment by patrickhulce** [comment]
The DevTools Lighthouse version can be found at the bottom of every report, Chrome 78 is 5.2.0. 
![image](https://user-images.githubusercontent.com/2301202/70541320-6da0ff00-1b2c-11ea-93a7-b9729b423ed3.png)



The programmatic one in this case is more reflective of the performance because it contains the fix for adjusting score weights.

**Comment by patrickhulce** [comment]
That's not the right report @wolvriene, we already know that programmatic is 5.6.0. The DevTools report could not possibly be produced by `HeadlessChrome` and if you're using Chrome 78, the Lighthouse version is 5.2.0.

> Also, Could you please provide more specific about score weights and how it actually works.

There are different score curves for desktop and mobile, the mobile curve is more forgiving.  When you test as a desktop device, in 5.6.0 it uses the desktop curve for the score. In older versions (i.e. 5.2.0), the desktop curve was not enforced properly and so the more forgiving curve was used instead which resulted in erroneously higher scores.

https://github.com/GoogleChrome/lighthouse/blob/4c4f48817c2ea46071c97d3567e728cc42613b66/lighthouse-core/audits/metrics/first-contentful-paint.js#L37-L52


https://github.com/GoogleChrome/lighthouse/blob/4c4f48817c2ea46071c97d3567e728cc42613b66/lighthouse-core/audits/metrics/first-contentful-paint.js#L64-L65

**Comment by wolvriene** [comment]
@patrickhulce 

Thanks for the update 
Yes in DevTool version is 5.2.0, Apologies for Incorrect screenshot, mistakenly i have uploaded it
I will more look into score weights and will close ticket shortly.

**Comment by wolvriene** [comment]
@patrickhulce 

I have updated chrome and current Version is 79.0.3945.79, lighthouse version is **5.5.0**

Does lighthouse version is **5.5.0** is having  fix for adjusting score weights.

Also i Checked with Chrome Canary Version Version 81.0.3992.4, lighthouse version is **5.7.0**.
Assuming it is having fix for adjusting score weights.But it does not have throttling option.

Is there any way i can compare **programmatic lighthouse results** with **DevTools lighthouse results** and conclude the same.

As there is huge difference between the performance score between them.

**Comment by patrickhulce** [comment]
> Is there any way i can compare programmatic lighthouse results with DevTools lighthouse results and conclude the same.

tl;dr - Yes but not with the settings you have.

First of all, to be clear, the metric results themselves should still be comparable between your two reports, it's just the scoring curve we're talking about here. The only versions of Lighthouse in DevTools that support the score curve adjustments have also removed the throttling setting, so you must use simulated throttling from the command line to see similar scores.

FWIW, that specific [commit](https://github.com/GoogleChrome/lighthouse/pull/9911) I'm referring to might not have reached DevTools yet either, so it might take a bit of time to reach Canary.

**Comment by wolvriene** [comment]
@patrickhulce 

Please correct me if i am wrong, so only **Performance score will be affected by score** curve, not other metrics for light house such as _First Contentful Paint , speed index , Time to Interactive etc_.

**Comment by patrickhulce** [comment]
> Performance score will be affected by score curve, not other metrics for light house such as First Contentful Paint , speed index , Time to Interactive etc.

Correct, the raw metric values are unaffected by the score curve. The score curve is only for the performance score.
